tests/ETHTornado.test.ts

```diff
@@ -2,21 +2,23 @@ import { assert, expect } from "chai";
 import { ETHTornado__factory, Verifier__factory, ETHTornado } from "../types/";
 
 import { ethers } from "hardhat";
-import { ContractFactory, BigNumber, BigNumberish } from "ethers";
+import { Contract, ContractFactory, BigNumber, BigNumberish } from "ethers";
 // @ts-ignore
-import { createCode, generateABI } from "circomlib/src/poseidon_gencontract";
+import { poseidonContract, buildPoseidon } from "circomlibjs";
 // @ts-ignore
-import poseidon from "circomlib/src/poseidon";
 import { MerkleTree, Hasher } from "../src/merkleTree";
 // @ts-ignore
 import { groth16 } from "snarkjs";
 import path from "path";
 
+import { readFileSync } from "fs";
+
 const ETH_AMOUNT = ethers.utils.parseEther("1");
 const HEIGHT = 20;
 
-function poseidonHash(inputs: BigNumberish[]): string {
+function poseidonHash(poseidon: any, inputs: BigNumberish[]): string {
     const hash = poseidon(inputs.map((x) => BigNumber.from(x).toBigInt()));
+    // pad zero to make it 32 bytes, so that the output can be taken as a bytes32 contract argument
     const bytes32 = ethers.utils.hexZeroPad(
         BigNumber.from(hash).toHexString(),
         32
```

Comment:
I tried another round. This can cosistently pass the test
```suggestion
    // Make the number within the field size
    const hashStr = poseidon.F.toString(hash);
    // Make it a valid hex string
    const hashHex = BigNumber.from(hashStr).toHexString();
    // pad zero to make it 32 bytes, so that the output can be taken as a bytes32 contract argument
    const bytes32 = ethers.utils.hexZeroPad(hashHex, 32);
```